pkg/services/ngalert/eval/eval.go

```diff
@@ -0,0 +1,184 @@
+// Package eval executes the condition for an alert definition, evaluates the condition results, and
+// returns the alert instance states.
+package eval
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/data"
+	"github.com/grafana/grafana-plugin-sdk-go/genproto/pluginv2"
+	"github.com/grafana/grafana/pkg/models"
+	"github.com/grafana/grafana/pkg/plugins"
+	"github.com/grafana/grafana/pkg/tsdb"
+)
+
+// Condition contains backend expressions and queries and the RefID
+// of the query or expression that will be evaluated.
+type Condition struct {
+	RefID string `json:"refId"`
+
+	QueriesAndExpressions []backend.DataQuery `json:"queriesAndExpressions"`
+}
+
+// ExecutionResults contains the unevaluated results from executing
+// a condition.
+type ExecutionResults struct {
+	AlertDefinitionID int64
+
+	Error error
+
+	Results data.Frames
+}
+
+// Results is a slice of evaluated alert instances states.
+type Results []Result
+
+// Result contains the evaluated state of an alert instance
+// identified by its labels.
+type Result struct {
+	Instance data.Labels
+	State    State // Enum
+}
+
+// State is an enum of the evaluation state for an alert instance.
```

Comment:
Don''t have to be public:
```suggestion
// result contains the evaluated state of an alert instance
// identified by its labels.
type result struct {
	Instance data.Labels
	State    state // Enum
}

// state is an enum of the evaluation state for an alert instance.
```